packages/integration-tests/src/test-e2e/backend_output.test.ts

```diff
@@ -84,6 +88,31 @@ void describe(
       await testProject.deploy(branchBackendIdentifier, sharedSecretsEnv);
       await testProject.assertPostDeployment(branchBackendIdentifier);
 
+      // assert storage access paths are correct in stack outputs
+      const outputsObject = JSON.parse(
+        await fsp.readFile(
+          path.join(testProject.projectDirPath, 'amplify_outputs.json'),
+          'utf-8'
+        )
+      );
+      assert.ok(
+        isMatch(outputsObject.storage.buckets[0].paths, {
+          'public/*': {
+            guest: ['get', 'list'],
+            authenticated: ['get', 'list', 'write'],
+            groupsAdmins: ['get', 'list', 'write', 'delete'],
+          },
+          'protected/*': {
+            authenticated: ['get', 'list'],
+            groupsAdmins: ['get', 'list', 'write', 'delete'],
+          },
+          'protected/${cognito-identity.amazonaws.com:sub}/*': {
+            // eslint-disable-next-line spellcheck/spell-checker
+            entityidentity: ['get', 'list', 'write', 'delete'],
+          },
+        })
+      );
```

Comment:
Is this the right place for this assertion ? I recall we use this test to assure that we can read outputs from project deployed with N-1 version/commit.

I'd rather see it in `packages/integration-tests/src/test-project-setup/data_storage_auth_with_triggers.ts` post deployment section.

Reply:
Oh right, I forgot the intention of this e2e test.

I'll add a comment at the top explaining what this test is for to prevent confusion in the future and move this assertion to post deployment you mentioned.